mempool/cat/pending_test.go

```diff
@@ -36,36 +38,52 @@ func TestPendingSeenTracker(t *testing.T) {
 			},
 		},
 		{
-			name: "deduplicate and append peer ids",
+			name: "re-adding same tx keeps first peer",
 			run: func(t *testing.T, tracker *pendingSeenTracker) {
 				key := txKey("tx2")
 				tracker.add(signer, key, 2, 5)
 				tracker.add(signer, key, 2, 5) // duplicate peer
-				tracker.add(signer, key, 2, 7) // new peer
+				tracker.add(signer, key, 2, 7) // different peer, should be ignored
 
 				entry := tracker.entriesForSigner(signer)[0]
-				require.ElementsMatch(t, []uint16{5, 7}, entry.peerIDs())
+				require.Equal(t, []uint16{5}, entry.peerIDs())
 			},
 		},
 		{
-			name:  "enforces per-signer limit and evicts oldest",
-			limit: 2,
+			name: "adding beyond nominal limit retains entries",
 			run: func(t *testing.T, tracker *pendingSeenTracker) {
 				key1 := txKey("a")
 				key2 := txKey("b")
 				key3 := txKey("c")
 
 				tracker.add(signer, key1, 1, 1)
 				tracker.add(signer, key2, 2, 2)
-				tracker.add(signer, key3, 3, 3) // should evict key1
+				tracker.add(signer, key3, 3, 3)
 
 				entries := tracker.entriesForSigner(signer)
-				require.Len(t, entries, 2)
-				require.Equal(t, []types.TxKey{key2, key3}, []types.TxKey{entries[0].txKey, entries[1].txKey})
+				require.Len(t, entries, 3)
+				require.Equal(t, []types.TxKey{key1, key2, key3}, []types.TxKey{entries[0].txKey, entries[1].txKey, entries[2].txKey})
 
 				require.Nil(t, tracker.entriesForSigner(otherSigner))
 			},
 		},
+		{
+			name:  "enforces per-signer limit",
+			limit: 2,
+			run: func(t *testing.T, tracker *pendingSeenTracker) {
+				key1 := txKey("a")
+				key2 := txKey("b")
+				key3 := txKey("c")
+
+				tracker.add(signer, key1, 1, 1)
+				tracker.add(signer, key2, 2, 2)
+				tracker.add(signer, key3, 3, 3)
+
+				entries := tracker.entriesForSigner(signer)
+				require.Len(t, entries, 2)
+				require.Equal(t, []types.TxKey{key1, key2}, []types.TxKey{entries[0].txKey, entries[1].txKey})
+			},
+		},
```

Comment:
If you add them out of order i.e. 1, 3, 2. Would it still keep sequences 1 and 2 and remove the tx with sequence 3? Maybe something to also unit test